gplog/gplog.go

```diff
@@ -265,63 +347,66 @@ func Progress(s string, v ...interface{}) {
 		_ = logger.logFile.Output(1, message)
 	}
 	if logger.shellVerbosity >= LOGVERBOSE {
-		message = GetLogPrefix("DEBUG") + fmt.Sprintf(s, v...)
+		message = GetShellLogPrefix("DEBUG") + fmt.Sprintf(s, v...)
 		_ = logger.logStdout.Output(1, message)
 	}
 }
 
 func Verbose(s string, v ...interface{}) {
 	logMutex.Lock()
 	defer logMutex.Unlock()
-	if logger.fileVerbosity >= LOGVERBOSE || logger.shellVerbosity >= LOGVERBOSE {
+	if logger.fileVerbosity >= LOGVERBOSE {
 		message := GetLogPrefix("DEBUG") + fmt.Sprintf(s, v...)
-		if logger.fileVerbosity >= LOGVERBOSE {
-			_ = logger.logFile.Output(1, message)
-		}
-		if logger.shellVerbosity >= LOGVERBOSE {
-			_ = logger.logStdout.Output(1, message)
-		}
+		_ = logger.logFile.Output(1, message)
+	}
+	if logger.shellVerbosity >= LOGVERBOSE {
+		message := GetShellLogPrefix("DEBUG") + fmt.Sprintf(s, v...)
+		_ = logger.logStdout.Output(1, message)
 	}
 }
 
 func Debug(s string, v ...interface{}) {
 	logMutex.Lock()
 	defer logMutex.Unlock()
-	if logger.fileVerbosity >= LOGDEBUG || logger.shellVerbosity >= LOGDEBUG {
+	if logger.fileVerbosity >= LOGDEBUG {
 		message := GetLogPrefix("DEBUG") + fmt.Sprintf(s, v...)
-		if logger.fileVerbosity >= LOGDEBUG {
-			_ = logger.logFile.Output(1, message)
-		}
-		if logger.shellVerbosity >= LOGDEBUG {
-			_ = logger.logStdout.Output(1, message)
-		}
+		_ = logger.logFile.Output(1, message)
+	}
+	if logger.shellVerbosity >= LOGDEBUG {
+		message := GetShellLogPrefix("DEBUG") + fmt.Sprintf(s, v...)
+		_ = logger.logStdout.Output(1, message)
 	}
 }
 
 func Error(s string, v ...interface{}) {
 	logMutex.Lock()
 	defer logMutex.Unlock()
-	message := GetLogPrefix("ERROR") + fmt.Sprintf(s, v...)
 	errorCode = 1
+	message := GetLogPrefix("ERROR") + fmt.Sprintf(s, v...)
 	_ = logger.logFile.Output(1, message)
-	_ = logger.logStderr.Output(1, message)
+	message = GetShellLogPrefix("ERROR") + fmt.Sprintf(s, v...)
+	_ = logger.logStderr.Output(1, Colorize(RED, message))
 }
 
 func Fatal(err error, s string, v ...interface{}) {
 	logMutex.Lock()
 	defer logMutex.Unlock()
-	message := GetLogPrefix("CRITICAL")
 	errorCode = 2
+	suffix := ""
 	stackTraceStr := ""
 	if err != nil {
-		message += fmt.Sprintf("%v", err)
+		suffix += fmt.Sprintf("%v", err)
 		stackTraceStr = formatStackTrace(errors.WithStack(err))
 		if s != "" {
-			message += ": "
+			suffix += ": "
 		}
 	}
-	message += strings.TrimSpace(fmt.Sprintf(s, v...))
+	suffix += strings.TrimSpace(fmt.Sprintf(s, v...))
+	message := GetLogPrefix("CRITICAL") + suffix
```

Comment:
The variable names feel a little strange to me here building up most things as `suffix`. I think it might feel more clear to me to build up the content as `message` and then create a new variable `logMessage` where you prepend the LogPrefix and `shellMessage` where you prepend the shell prefix? Or technically you could append them together when you do output like 

`_ = logger.logFile.Output(1, GetLogPrefix("CRITICAL")+message+stackTraceStr)` and
`_ = logger.logFile.Output(1, GetShellLogPrefix("CRITICAL"+message+stackTraceStr)`

Reply:
renamed `suffix` -> `message` and `message` -> `fullMessage`